test/acceptance/tests/terminating-gateway/terminating_gateway_test.go

```diff
@@ -11,61 +12,148 @@ import (
+			t.Log("registering the external service")
+			_, err := consulClient.Catalog().Register(&api.CatalogRegistration{
+				Node:     "legacy_node",
+				Address:  "static-server",
+				NodeMeta: map[string]string{"external-node": "true", "external-probe": "true"},
+				Service: &api.AgentService{
+					ID:      "static-server",
+					Service: "static-server",
+					Port:    80,
+				},
+			}, &api.WriteOptions{})
+			require.NoError(t, err)
+
+			// If ACLs are enabled we need to update the token of the terminating gateway
+			// with service:write permissions to the static-server service
+			// so that it can can request Connect certificates for it.
+			if c.secure {
+				// Create a write policy for the static-server
+				_, _, err := consulClient.ACL().PolicyCreate(&api.ACLPolicy{
+					Name:  "static-server-write-policy",
+					Rules: staticServerPolicyRules,
+				}, nil)
+				require.NoError(t, err)
+
+				// Get the terminating gateway token
+				tokens, _, err := consulClient.ACL().TokenList(nil)
+				require.NoError(t, err)
+				var termGwTokenID string
+				for _, token := range tokens {
+					if strings.Contains(token.Description, "terminating-gateway-terminating-gateway-token") {
+						termGwTokenID = token.AccessorID
+						break
+					}
+				}
+				termGwToken, _, err := consulClient.ACL().TokenRead(termGwTokenID, nil)
+				require.NoError(t, err)
+
+				// Add policy to the token and update it
+				termGwToken.Policies = append(termGwToken.Policies, &api.ACLTokenPolicyLink{Name: "static-server-write-policy"})
+				_, _, err = consulClient.ACL().TokenUpdate(termGwToken, nil)
+				require.NoError(t, err)
+			}
+
+			// Create the config entry for the terminating gateway
+			t.Log("creating config entry")
+			created, _, err := consulClient.ConfigEntries().Set(&api.TerminatingGatewayConfigEntry{
+				Kind:     api.TerminatingGateway,
+				Name:     "terminating-gateway",
+				Services: []api.LinkedService{{Name: "static-server"}},
+			}, nil)
+			require.NoError(t, err)
+			require.True(t, created, "config entry failed")
+
+			// Deploy the static client
+			t.Log("deploying static client")
+			helpers.Deploy(t, ctx.KubectlOptions(), cfg.NoCleanupOnFailure, "fixtures/static-client.yaml")
+
+			// If ACLs are enabled, test that intentions prevent connections.
+			if c.secure {
+				// With the terminating gateway up, we test that we can make a call to it
+				// via the static-server. It should fail to connect with the
+				// static-server pod because of intentions.
+				t.Log("testing intentions prevent connections through the terminating gateway")
+				helpers.CheckStaticServerConnection(t,
+					ctx.KubectlOptions(),
+					"static-client",
+					false,
+					"http://localhost:1234")
+
+				// Now we create the allow intention.
+				t.Log("creating static-client => static-server intention")
+				_, _, err = consulClient.Connect().IntentionCreate(&api.Intention{
+					SourceName:      "static-client",
```

Comment:
Constantize "static-server" and "static-client"? We seem to use them all over the place

Reply:
Do you mean make them exported constants to be shared by all tests? or just within this test?

Reply:
That is an excellent question 😛 
I think per file is alright for now. We can think of global constants if we see an explosion of them.

Reply:
Constantized!